store/snap_remote_repo.go

```diff
@@ -232,6 +257,111 @@ func (s *SnapUbuntuStoreRepository) checkStoreResponse(resp *http.Response) {
 	}
 }
 
+func getRequiresPurchase(prices map[string]float64, purchases purchaseList) bool {
+	// if the snap is free, then it doesn't need purchasing
+	if len(prices) == 0 {
+		return false
+	}
+
+	for _, purchase := range purchases {
+		// if the purchase is not an in-app purchase
+		if purchase.ItemSKU == "" {
+			return false
+		}
+	}
+
+	// the snap is not free, and we couldn't find a purchase
+	return true
+}
+
+func (s *SnapUbuntuStoreRepository) getPurchases(name string, auther Authenticator) (purchaseList, error) {
+	var purchases purchaseList
+
+	// only build the request if we have an Authenticator
+	if auther != nil {
+		purchasesURL, err := s.purchasesURI.Parse(name + "/")
+		if err != nil {
+			return nil, err
+		}
+
+		q := purchasesURL.Query()
+		q.Set("include_item_purchases", "true")
+		purchasesURL.RawQuery = q.Encode()
+
+		req, err := http.NewRequest("GET", purchasesURL.String(), nil)
+		if err != nil {
+			return nil, err
+		}
+
+		s.applyUbuntuStoreHeaders(req, "application/json", auther)
+
+		resp, err := s.client.Do(req)
+		if err != nil {
+			return nil, err
+		}
+		defer resp.Body.Close()
+
+		switch {
+		case resp.StatusCode == http.StatusNotFound:
+			break
```

Comment:
Isn't that a 404 on an API endpoint we assumed should exist for things to work?  That looks like a real error we should report rather than silently assume it's okay.